utils/questGen.py

```diff
@@ -77,3 +72,104 @@ def pokemonname(id):
     with open('pokemon.json') as f:
         mondata = json.load(f)
     return mondata[str(int(id))]["name"]
+
+def questtask(typeid, condition, target):
+    with open('utils/quest/pokemonTypes.json') as f:
+        pokemonTypes = json.load(f)
+    with open('utils/quest/items.json') as f:
+        items = json.load(f)
+    throwTypes = {"10":"Nice", "11":"Great", "12":"Excellent", "13":"Curveball"}
+    arr = {}
+    arr['0'] = target
+    text = questtype(typeid)
+
+    if typeid == 4:
+        arr['wb'] = ""
+        arr['type'] = ""
+        text = "Catch {0} {type}Pokemon{wb}."
+        match_object = re.search(r"'pokemon_type': \[([0-9, ]+)\]", condition)
+        if match_object is not None:
+                pt = match_object.group(1).split(', ')
+                last = len(pt)
+                cur = 1
+                if last == 1:
+                    arr['type'] = pokemonTypes[pt[0]].title() + '-type '
+                else:
+                    for ty in pt:
+                        arr['type'] += ('or ' if last == cur else '') + pokemonTypes[ty].title() + ('-type ' if last == cur else '-, ')
+                        cur += 1
+        if re.search(r"'type': 3", condition) is not None:
+                arr['wb'] = " with weather boost"
+        match_object = re.search(r"'pokemon_ids': \[([0-9, ]+)\]", condition)
+        if match_object is not None:
+                pt = match_object.group(1).split(', ')
+                last = len(pt)
+                cur = 1
+                if last == 1:
+                    arr['poke'] = pokemonname[pt[0]]
+                text = 'Catch a {poke}.'
```

Comment:
Do you potentially need this too in case there are multiple pokemon?

`else:
        for ty in pt:
        arr['poke'] += ('or ' if last == cur else '') + pokemonname[ty]
         cur += 1            `